testing/reconciler_test.go

```diff
@@ -19,48 +19,57 @@ package testing
 import (
 	"context"
 	"testing"
+	"time"
 
 	"k8s.io/apimachinery/pkg/runtime"
 	"reconciler.io/runtime/reconcilers"
+	rtime "reconciler.io/runtime/time"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
-func TestReconcilerTestCase_Run(t *testing.T) {
+func TestReconcilerTestCase_Now(t *testing.T) {
+	start := time.Now()
+
 	type testCase struct {
-		name      string
-		verify    func(*testing.T, *ReconcilerTestCase, *testCase)
-		verifyRan bool
+		name   string
+		rtc    *ReconcilerTestCase
+		verify func(*testing.T, context.Context)
 	}
 	tests := []*testCase{
 		{
 			name: "Now defaults to time.Now()",
-			verify: func(t *testing.T, rtc *ReconcilerTestCase, tc *testCase) {
-				tc.verifyRan = true
-				if rtc.Now.IsZero() {
+			rtc:  &ReconcilerTestCase{},
+			verify: func(t *testing.T, ctx context.Context) {
+				now := rtime.RetrieveNow(ctx)
+				// compare with a range to allow for time skew
+				// test must complete within 1 hour of wall clock time
+				if now.Before(start.Add(-1*time.Second)) || now.After(start.Add(time.Hour)) {
```

Comment:
```suggestion
				// compare with a range to allow for time skew
				// test must complete within 2 seconds of wall clock time
				if now.Before(start.Add(-1*time.Second)) || now.After(start.Add(2*time.Second)) {
```

1 hour seems overly large tbh. This is a unit test so I expect this to take very little time – wdyt?

Reply:
For regular runs I agree. However, if debugging this or another test the elapsed time can easily exceed a few seconds. An hour feels like a fair balance.

Reply:
Alright, agreed.